src/kv/MemDB.cc

```diff
@@ -397,7 +401,7 @@ void MemDB::MDBWholeSpaceIteratorImpl::fill_current()
 
 bool MemDB::MDBWholeSpaceIteratorImpl::valid()
 {
-  if (m_key_value.first.empty()) {
+  if (!(*is_valid) || m_key_value.first.empty()) {
```

Comment:
@chhabaramesh @liewegas Thanks for your comments.

I'm not sure if bluestore will retry to get a valid iterator before, so I just add a bool. Then I will fix it with seqno:)